File: codetree/trail2/palindrome 여부 판단하기/determine-whether-palindrome-is-present.py
import os
import sys

file_path = os.path.join(os.path.dirname(__file__), "input_test.txt")

if os.path.exists(file_path):
    sys.stdin = open(file_path, "r", encoding="utf-8")


# 📥 Input
def get_input_data() -> str:
    word = input().strip()

    return word


# ⚙️ Logic
def solution(word: str) -> str:
    if word == word[::-1]:
        return "Yes"
    else:
        return "No"


# 🚀 Run Program
if __name__ == "__main__":
    print(solution(get_input_data()))

File: codetree/trail2/palindrome 여부 판단하기/determine-whether-palindrome-is-present.ts
declare var require: any;
const fs: any = require("fs");

const filePath: string = fs.existsSync("./input_test.txt")
    ? "./input_test.txt"
    : "/dev/stdin";

const input: string[] = fs.readFileSync(filePath, "utf-8").trim().split(/\n+/);

/* 📥 Input */
const getInputData = (): string => {
    const word = input[0]
  
    return word
};

/* ⚙️ Logic */
const solution = (word: string): string => {
    for (let i = 0; i < word.length; i++) {
        if (word[i] !== word[word.length - i - 1]) {
            return "No";
        }
    }
    return "Yes";
};

/* 🚀 Run Program */
(() => {
    console.log(solution(getInputData())); 
})();